internal/metamorph/store/postgresql/postgres.go

```diff
@@ -642,7 +635,8 @@ func (p *PostgreSQL) UpdateDoubleSpend(ctx context.Context, updates []store.Upda
 			) AS bulk_query
 			WHERE metamorph.transactions.hash=bulk_query.hash
 				AND metamorph.transactions.status <= bulk_query.status
-				AND LENGTH(metamorph.transactions.competing_txs) < LENGTH(bulk_query.competing_txs)
+				AND (metamorph.transactions.competing_txs IS NULL
+						OR LENGTH(metamorph.transactions.competing_txs) < LENGTH(bulk_query.competing_txs))
```

Comment:
That means that the competing txs will only be updated if the updates contain more competing txs than the existing one, right?

What if there is just one more competing tx that should get added which is not among the existing ones, shouldn't that also be updated? 🤔

Reply:
If you look at line `678` and then at line `693`, we first get already existing competing transactions from the database and then we merge them with incoming transactions.

Therefore:
* if at least one new transaction is coming that was not previously there, it will be merged with already existing transactions and updated in the record
* if no new transactions are coming, or if the incoming transactions are already existing in the db, we will avoid the update to save resources